workflow_generator/ecflow_setup/ecflow_definitions.py

```diff
@@ -741,140 +781,104 @@ def add_task_triggers(self, task, triggers, suite_array):
+                                for state_item in state:
+                                    self.add_trigger(trigger_name, node_name,
+                                                     suite=suite,
+                                                     suite_array=suite_array,
+                                                     event=event,
+                                                     state=state_item,
+                                                     operand=operand)
+                            else:
+                                self.add_trigger(trigger_name, node_name,
+                                                 suite=suite,
+                                                 suite_array=suite_array,
+                                                 event=event,
+                                                 operand=operand)
                     else:
-                        process_trigger(triggerTaskNode.get_name(), triggerTaskNode, task_name,
-                                        task_loop_index, total_tasks, task_number)
-                task_loop_index += 1
-        else:
-            for trigger in triggers:
-                triggerTaskNode = ecfTriggerNode(trigger)
-                if triggerTaskNode.is_loop():
-                    try:
-                        if triggerTaskNode.has_max_value():
-                            for trigger_flag in triggerTaskNode.get_range():
-                                trigger_name = f"{triggerTaskNode.get_full_name(trigger_flag)}"
-                                process_trigger(trigger_name, triggerTaskNode, task)
+                        if trigger_node.has_state():
+                            state = trigger_node.get_state()
+                            if not isinstance(state, list):
+                                state = [state]
+                            for state_item in state:
+                                self.add_trigger(trigger_name, node_name,
+                                                 suite=suite,
+                                                 suite_array=suite_array,
+                                                 state=state_item,
+                                                 operand=operand)
                         else:
-                            raise ConfigurationError
-                    except ConfigurationError:
-                        print(f"Task: {task} - Looping mechanism called "
-                              "without max value in a non looped task.")
-                        sys.exit(1)
-                else:
-                    process_trigger(triggerTaskNode.get_name(), triggerTaskNode, task)
+                            self.add_trigger(trigger_name, node_name,
+                                             suite=suite,
+                                             suite_array=suite_array,
+                                             operand=operand)
 
 
 class ecfNode():
```

Comment:
All these class names should be camel case and start with a capital letter. This is something I missed last time. So:

EcflowSuite (or maybe ECFlowSuite)
EcfNode
EcFamilyNode

Reply:
I forgot to respond to this that I had updated the camel case appropriately with a follow on commit.